platform/darwin/http_request_nsurl.mm

```diff
@@ -154,13 +154,15 @@
     @autoreleasepool {
         
         NSMutableString *url = [NSMutableString stringWithString:@(resource.url.c_str())];
-        if ([[NSUserDefaults standardUserDefaults] integerForKey:@"MGLMapboxAccountType"] == 0) {
-            if ([url rangeOfString:@"?"].location == NSNotFound) {
-                [url appendString:@"?"];
-            } else {
-                [url appendString:@"&"];
+        if ([url rangeOfString:@"mapbox.com"].location != NSNotFound) {
```

Comment:
What if “mapbox.com” appears elsewhere in the URL besides the hostname? `-[NSURL host]` is much better suited for this kind of comparison. If that turns out to be unperformant for some reason, there’s also `CFURLCopyHostName()`.